Breadcrumbs.py

```diff
@@ -32,80 +38,169 @@ def get_row_indentation(points, view, tab_size, limit=1e20):
+
+  breadcrumbs.reverse()
+  lengths = [len(breadcrumb) for breadcrumb in breadcrumbs]
+  sorted_lengths = sorted(lengths)
+  previous_length = 0
+  number_of_characters_left  = total_breadcrumbs_length_limit - len(lengths) * len(separator)
+  for (number_of_shorter, current_length) in enumerate(sorted_lengths):
+    if previous_length < current_length:
+      previous_length = current_length
+      number_of_not_shorter = len(breadcrumbs) - number_of_shorter
+      if number_of_characters_left < current_length * number_of_not_shorter:
+        length_of_short_trim = number_of_characters_left // number_of_not_shorter
+        length_of_long_trim = length_of_short_trim + 1
+        number_of_long_trimmed = number_of_characters_left % number_of_not_shorter
+        number_of_short_trimmed = number_of_not_shorter - number_of_long_trimmed
+        for (index_of_breadcrumb, breadcrum_length) in enumerate(lengths):
+          if current_length <= breadcrum_length:
+            if 0 < number_of_short_trimmed:
+              length_of_trim = length_of_short_trim
+              number_of_short_trimmed -= 1
+            else:
+              length_of_trim = length_of_long_trim
+            breadcrumbs[index_of_breadcrumb] = breadcrumbs[index_of_breadcrumb][0:length_of_trim]
+        break
+    number_of_characters_left -= current_length
+
+  return ''.join(breadcrumbs+[''])
+
+
+stylesheet = '''
+  <style>
+    div.phantom-arrow {
+      border-top: 0.4rem solid transparent;
+      border-left: 0.5rem solid color(var(--bluish) blend(var(--background) 30%));
+      width: 0;
+      height: 0;
+    }
+    div.phantom {
+      padding: 0.4rem 0 0.4rem 0.7rem;
+      margin: 0 0 0.2rem;
+      border-radius: 0 0.2rem 0.2rem 0.2rem;
+      background-color: color(var(--bluish) blend(var(--background) 30%));
+    }
+    div.phantom span.message {
+      padding-right: 0.7rem;
+    }
+    div.phantom a {
+      text-decoration: inherit;
+    }
+    div.phantom a.close {
+      padding: 0.35rem 0.7rem 0.45rem 0.8rem;
+      position: relative;
+      bottom: 0.05rem;
+      border-radius: 0 0.2rem 0.2rem 0;
+      font-weight: bold;
+    }
+    html.dark div.phantom a.close {
+      background-color: #00000018;
+    }
+    html.light div.phantom a.close {
+      background-color: #ffffff18;
+    }
+  </style>
+'''
+
+
+template = '''
+  <body>
+    {stylesheet}
+    <div class="phantom-arrow"></div>
+    <div class="phantom">
+      <span class="message">
+        <strong>Breadcrumbs:</strong> {breadcrumbs}
+        <a class="close" href="close">''' + chr(0x00D7) + '''</a>
+      </span>
+    </div>
+  </body>
+'''
+
+
+class BreadcrumbsCommand(sublime_plugin.EventListener):
+
+  def on_selection_modified_async(self, view):
+    if settings.get('breadcrumbs_statusbar', True):
+      view.set_status('breadcrumbs', make_breadcrumbs(view))
+    else:
+      view.erase_status('breadcrumbs')
+
+
+class BreadcrumbsPhantomCommand(sublime_plugin.TextCommand):
```

Comment:
> I believe we should document both of these possibilities in the README.md.

That's a good idea. I don't think it's a good idea for packages to claim keyboard shortcuts, but how to do that should be in the README. 
There is also a context menu by the way, if you right-click in your code the command is there too.

> The way I imagine using it myself is that I want the breadcrumbs to be always visible, and update their content and position as I move my cursor(s)

That would be an awesome mode to be able to toggle on and off!

> Also, it would be great to be able to close the Phantom(s) using a keyboard, but I do not know how to do that yet.

Same here. I'll try to find out how `exec.py` does this and see if I can reproduce it.